econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -1571,14 +1579,46 @@ def individualized_policy(self, Xtest, feature_index, *, n_rows=None, treatment_
             zeros = np.zeros((est.shape[0], 1))
             all_effs = np.hstack([zeros, est])
             eff_ind = np.argmax(all_effs, axis=1)
-            all_eff_lbs = np.hstack([zeros, est_lb])
-            all_eff_ubs = np.hstack([zeros, est_ub])
             treatment_arr = np.array([result.feature_baseline] + [lvl for lvl in result.feature_levels], dtype=object)
             rec = treatment_arr[eff_ind]
-            eff_ind = eff_ind.reshape(-1, 1)
-            eff = np.take_along_axis(all_effs, eff_ind, 1).reshape(-1)
-            eff_lb = np.take_along_axis(all_eff_lbs, eff_ind, 1).reshape(-1)
-            eff_ub = np.take_along_axis(all_eff_ubs, eff_ind, 1).reshape(-1)
+            # we need to call effect_inference to get the correct CI between the two treatment options
+            effect = result.estimator.effect_inference(Xtest, T0=orig_df['Current treatment'], T1=rec)
+            # we now need to construct the delta in the cost between the two treatments and translate the effect
+            current_treatment = orig_df['Current treatment'].values
+            if np.ndim(treatment_costs) >= 2:
```

Comment:
I don't love all of the array dimensionality munging, weird boolean indexing, etc. here and further below, but I spent a while trying to simplify it without much luck so maybe there's no easy way around it.  Ideally add a TODO to take a closer look at whether we can simplify some of the logic in a later batch of work.